File: tracks/java/src/ArmstrongNumbers.java
/**
 * The {@code ArmstrongNumbers} class provides a method to determine whether a
 * given integer
 * is an Armstrong number or not. An Armstrong number (also known as a
 * narcissistic number,
 * pluperfect digit, or Plus Perfect number) is a number that is the sum of its
 * own digits
 * each raised to the power of the number of digits.
 */
public class ArmstrongNumbers {

    /**
     * Default constructor for ArmstrongNumbers.
     * This constructor doesn't perform any special operations and is provided
     * to create instances of this class if needed.
     */
    public ArmstrongNumbers() {

    }

    /**
     * Determines if a given integer is an Armstrong number.
     * An Armstrong number is a number that is equal to the sum of its own digits
     * each raised to
     * the power of the number of digits. For example, 153 is an Armstrong number
     * since
     * 1^3 + 5^3 + 3^3 = 153.
     *
     * @param numberToCheck the integer number to be checked.
     * @return {@code true} if the given number is an Armstrong number,
     *         {@code false} otherwise.
     */
    boolean isArmstrongNumber(int numberToCheck) {

        char[] numArray = String.valueOf(numberToCheck).toCharArray();
        int len = numArray.length;
        int sum = 0;

        for (int i = 0; i < len; i++) {
            int digit = numArray[i] - '0';
            sum += Math.pow(digit, len);
        }

        return sum == numberToCheck;
    }

}

File: tracks/python/armstrong_numbers.py
"""
This module is dedicated to operations related to Armstrong numbers.

An Armstrong number (also known as a narcissistic number) is a number that is equal to the sum of its own digits each raised to the power of the number of digits. The module provides functionality to check whether a given number is an Armstrong number. This can be useful in various mathematical and educational applications where number theory is involved.

Functions:
- is_armstrong_number(number): Determines if a given number is an Armstrong number.

Example usage:
>>> is_armstrong_number(153)
True
>>> is_armstrong_number(123)
False
"""


def is_armstrong_number(number):
    """
    Check if a given number is an Armstrong number.

    An Armstrong number, also known as a narcissistic number, is a number that is equal to the sum of its own digits each raised to the power of the number of digits. This function determines if the given number falls into this category.

    :param number: int - The number to be checked.
    :return: bool - Returns True if the number is an Armstrong number, False otherwise.
    """

    num_str = str(number)
    result = 0
    for n in num_str:
        result += int(n) ** len(num_str)
    return number == result
